src/link_prober/LinkProber.cpp

```diff
@@ -232,18 +232,23 @@ void LinkProber::handleUpdateEthernetFrame()
 //
 void LinkProber::handleSendSwitchCommand()
 {
-    IcmpPayload *icmpPayload = reinterpret_cast<IcmpPayload *> (
-        mTxBuffer.data() + sizeof(ether_header) + sizeof(iphdr) + sizeof(icmphdr)
-    );
-    icmpPayload->command = htonl(static_cast<uint32_t> (Command::COMMAND_SWITCH_ACTIVE));
+    resetTxBufferTlv();
+    appendTlvCommand(Command::COMMAND_SWITCH_ACTIVE);
+    appendTlvSentinel();
 
+    size_t totalPayloadSize = mTxPacketSize - mPacketHeaderSize;
+    iphdr *ipHeader = reinterpret_cast<iphdr *> (mTxBuffer.data() + sizeof(ether_header));
     icmphdr *icmpHeader = reinterpret_cast<icmphdr *> (mTxBuffer.data() + sizeof(ether_header) + sizeof(iphdr));
-    computeChecksum(icmpHeader, sizeof(icmphdr) + sizeof(*icmpPayload));
+    computeChecksum(icmpHeader, sizeof(icmphdr) + totalPayloadSize);
+    ipHeader->tot_len = htons(sizeof(iphdr) + sizeof(icmphdr) + totalPayloadSize);
 
     sendHeartbeat();
 
-    icmpPayload->command = htonl(static_cast<uint32_t> (Command::COMMAND_NONE));
-    computeChecksum(icmpHeader, sizeof(icmphdr) + sizeof(*icmpPayload));
+    resetTxBufferTlv();
+    appendTlvSentinel();
```

Comment:
Do we only append TlvCommand when it's COMMAND_SWITCH_ACTIVE instead of having it in every heartbeat (COMMAND_NONE)? Shouldn't we have it in every heartbeat?

Reply:
Yes, currently the logic is to put command tlv only if there is need to send `COMMAND_SWITCH_ACTIVE`. Since now we remove the fixed command field from `IcmpPayload` and wrap it as a TLV, no `TlvCommand` is the same a `TlvCommand` with `COMMAND_NONE`.